chain/chain/src/store.rs

```diff
@@ -2133,7 +2148,23 @@ impl<'a> ChainStoreUpdate<'a> {
     ) -> Result<(), Error> {
         for part_id in 0..num_parts {
             let key = StatePartKey(sync_hash, shard_id, part_id).try_to_vec()?;
-            self.gc_col(DBCol::ColStateParts, &key);
+            self.gc_col(ColStateParts, &key);
+        }
+        Ok(())
+    }
+
+    pub fn gc_col_transaction(&mut self, tx_hash: CryptoHash) -> Result<(), Error> {
+        let mut refs = self.get_tx_refcount(&tx_hash)?;
+        debug_assert!(refs > 0);
```

Comment:
If we only want to crash in debug, the below should be `refs.saturating_sub(1)`.

Reply:
++